src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -859,6 +896,11 @@ def delete_connectedk8s(cmd, client, resource_group_name, cluster_name,
 
     logger.warning("This operation might take a while ...\n")
 
+    # Check if the cluster is of supported type for deletion
+    preview_cluster_resource = get_connectedk8s_2023_11_01(cmd, resource_group_name, cluster_name)
+    if preview_cluster_resource.kind == "ProvisionedCluster":
```

Comment:
>preview_cluster_resource.kind [](http://example.com/codeflow?start=7&length=29)

Check preview_cluter_resource is not null before using and convert kind to lowercase and compare against 'provisionedcluster'.